HIPs/incubation/web3j.md

```diff
@@ -0,0 +1,204 @@
+
+Web3j was created originally with the goal of addressing these needs for users. However, over time its scope grew to address some of the challenges with the overall development workflow when working with smart contracts on the JVM:
+
+- Scaffolding Solidity projects
+- Providing an integrated developer experience
+- A lack of unit and integration testing capabilities
+
+The specific components of Web3j and how it addresses these items are addressed in subsequent sections below.
+
+For further historical context on Web3j, you can read the [following post I wrote](https://web3perspectives.com/p/building-open-source-software-web3j) on the creation of the project earlier the year.
+
+# Status
+
+Proposal
+
+# Solution
+
+Web3j is made up of a number of components. It was designed to be as accessible as possible to developers, and the design of these components reflects this.
+
+In order to bring the greatest benefit to its users, the expectation is these components collectively would make up the Hyperledger Web3j project.
+
+These components are as follows:
+
+- The core Web3j library provides the Ethereum integration capabilities to support use cases that utilise the JSON-RPC protocol and smart contracts
+- Smart contract testing components that allow users to develop and test Solidity smart contracts within their JVM project workflow
+- Build plugins that integrate the Solidity compiler and Solidity library dependencies within the JVM project
+- A command line interface (CLI) to simplify working with Web3j
+
+These are discussed in further detail below.
+
+## Core Web3j library
+
+The [core Web3j library](https://github.com/web3j/web3j) was designed to be highly modular, so that users could easily incorporate the minimal parts of the library for their projects. This consideration was especially important in servicing the Android community.
+
+The notable modules include:
+
+- **rlp:** [Recursive Length Prefix (RLP)](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/) serialisation
+- **crypto:** cryptographic operations and key management 
+- **abi:** [Contract Application Binary Interface (ABI)](https://docs.soliditylang.org/en/latest/abi-spec.html) encoding and decoding
+- core: core Web3j components including the JSON-RPC implementation and abstractions for wallets
+- utils: utility functions including type conversions
+- **besu, geth, eea: Ethereum** client specific functionality
+- codegen: code generators for creating Java type wrappers for Solidity smart contracts
+- contracts: smart contract specific types
+
+## Smart Contract Testing
+
+The Web3j core library is complemented by the following projects that facilitate unit and integration testing of Solidity smart contracts.
+
+The [Web3j EVM](https://github.com/web3j/web3j-evm) project provides an embedded EVM that allows developers to write applications that simulate transactions taking place on the EVM. 
```

Comment:
Would this repository also be coming to Hyperledger?

Reply:
With listing these projects here, the intention is that they would come alongside Web3j.